File: leetcode/1759/solution.py
"""
https://leetcode.com/problems/count-number-of-homogenous-substrings/
1759. Count Number of Homogenous Substrings
"""


class Solution:
    def countHomogenous(self, s: str) -> int:
        total = count = 1
        MOD = 1000000007
        for i in range(1, len(s)):
            count = (count + 1) % MOD if s[i] == s[i - 1] else 1
            total = total + count
        return total % MOD

File: leetcode/1759/solution.go
/**
 * https://leetcode.com/problems/count-number-of-homogenous-substrings/
 * 1759. Count Number of Homogenous Substrings
 */
package main

func countHomogenous(s string) int {
	total, count := 1, 1
	MOD := 1_000_000_007
	for i := 1; i < len(s); i++ {
		if s[i] == s[i-1] {
			count = (count + 1) % MOD
		} else {
			count = 1
		}
		total = (total + count) % MOD
	}
	return total
}
